update_deps/propagate_updates.ps1

```diff
@@ -1,2 +1,509 @@
 # Copyright (c) Microsoft. All rights reserved.
 # Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+<#
+.SYNOPSIS
+
+Propagates dependency updates for git repositories.
+
+.DESCRIPTION
+
+Given a root repo and personal access tokens for Github and Azure Devops Services, this script \
+builds the dependency graph and propagates updates from the lowest level upto the \
+root repo by making PRs to each repo in bottom-up level-order.
+
+.PARAMETER root
+
+URL of the repository upto which updates must be propagated.
+
+.PARAMETER github_token
+
+Personal access token for Github
+
+.PARAMETER azure_token
+
+Personal access token for Azure Devops Services
+
+.INPUTS
+
+ignore.json: list of repositories that must be ignored for updates.
+order.json: if -resume flag is set, reads order in which repositories must be updated from order.json
+done.txt: if -resume flag is set, reads list of repositories to skip from done.txt
+
+.OUTPUTS
+
+Prints names of repositories that have been updated to done.txt
+
+.EXAMPLE
+
+PS> .\{PATH_TO_SCRIPT}\propagate_updates.ps1 -azure_token {token1} -github_token {token2} [-root {root_repo_url}] [-resume]
+#>
+
+
+param (
+    [string]$root="https://msazure.visualstudio.com/DefaultCollection/One/_git/Azure-MessagingStore", # url for repo upto which updates must be propagated
+    [Parameter(Mandatory=$true)][string]$github_token, # Github personal access token: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token
+    [Parameter(Mandatory=$true)][string]$azure_token, # Azure Devops Services personal access token: https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=azure-devops&tabs=preview-page
+    [switch]$resume = $false # flag to continue from checkpoint of previous run
```

Comment:
>[switch]$resume = $false # flag to continue from checkpoint of previous run [](http://example.com/codeflow?start=4&length=75)

Why is this needed?
I would think that whenever a new run is started if we go through all the repos and we find that no update needs to be done then there would be no PRs issued and the script would effectively resume where it left off.

Reply:
Removed in next iteration.